src/Components/Web.JS/src/Platform/Mono/MonoTypes.d.ts

```diff
@@ -29,3 +29,19 @@ declare namespace MONO {
 // mono_bind_static_method allows arbitrary JS data types to be sent over the wire. However we are
 // artifically limiting it to a subset of types that we actually use.
 declare type BoundStaticMethod = (...args: (string | number | null)[]) => (string | number | null);
+
+declare namespace BINDING {
+  function js_string_to_mono_string(jsString: string): Pointer;
+  function js_typed_array_to_array(array: Uint8Array): Pointer;
+  function js_typed_array_to_array<T>(array: Array<T>): Pointer;
+  function conv_string(dotnetString: System_String | null): string | null;
+}
+
+// We don't actually instantiate any of these at runtime. For perf it's preferable to
+// use the original 'number' instances without any boxing. The definitions are just
+// for compile-time checking, since TypeScript doesn't support nominal types.
+declare interface MethodHandle { MethodHandle__DO_NOT_IMPLEMENT: any }
+declare interface System_Object { System_Object__DO_NOT_IMPLEMENT: any }
+declare interface System_String extends System_Object { System_String__DO_NOT_IMPLEMENT: any }
+declare interface System_Array<T> extends System_Object { System_Array__DO_NOT_IMPLEMENT: any }
+declare interface Pointer { Pointer__DO_NOT_IMPLEMENT: any }
```

Comment:
I don't think any of these definitions (`System_Object` to `Pointer`) should be moved from `Platform.ts` into this file.

As long as we're keeping `Platform.ts`, then the dependencies should be from `MonoPlatform.ts` to `Platform.ts`, not the other way around.